edx_proctoring/tests/test_api.py

```diff
@@ -499,6 +500,233 @@ def test_remove_allowance_for_user(self):
         remove_allowance_for_user(student_allowance.proctored_exam.id, self.user_id, self.key)
         self.assertEqual(len(ProctoredExamStudentAllowance.objects.filter()), 0)
 
+    @ddt.data(
+        (
+            ADDITIONAL_TIME,
+            '30',
+            '30',
+            '30'
+        ),
+        (
+            TIME_MULTIPLIER,
+            '1.5',
```

Comment:
Can you please add non-digit test cases, like `abc` or `24bb`?

Reply:
I have unit tests for invalid values under the test_views.py file, should I also add such tests to test_api.py. As a follow-up, how should I determine which tests should be in either file or both?
`
    @ddt.data(        (            ADDITIONAL_TIME,            '-30'        ),        (            TIME_MULTIPLIER,            '-1.5'        ),        (            TIME_MULTIPLIER,            'invalid_value'        )    )    @ddt.unpack    def test_add_bulk_allowance_invalid_allowance_value(self, allowance_type, value):  # pylint: disable=invalid-name        """        Test to add bulk allowance with invalid allowance value        """        # Create exams.        user_list = self.create_batch_users(3)        user_id_list = [user.email for user in user_list]        exam1 = ProctoredExam.objects.create(            course_id='a/b/c',            content_id='test_content',            exam_name='Test Exam',            time_limit_mins=90,            is_active=True            )        exam2 = ProctoredExam.objects.create(            course_id='a/b/c',            content_id='test_content2',            exam_name='Test Exam2',            time_limit_mins=90,            is_active=True            )        exam_list = [exam1.id, exam2.id]        allowance_data = {            'exam_ids': exam_list,            'user_ids': user_id_list,            'allowance_type': allowance_type,            'value': value        }        response = self.client.put(            reverse('edx_proctoring:proctored_exam.bulk_allowance'),            json.dumps(allowance_data),            content_type='application/json'        )        self.assertEqual(response.status_code, 400)`